### PR TITLE
Feature/tile component

### PR DESCRIPTION
**Pull request by N00BKU**

### CODE REVIEW

File: src/components/TileComponents.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { render } from '../shared/dom';
+
+export const TileComponent = ({ title, image }) => {
+    return `
+    <div class="movie-tile">
+        <img src="${image}"/>
+        <p class='title'>${title}</p>
```
Comment (Veryyapeee) [inline src/components/TileComponents.js]: title is actually not a paragraph, it should be Hx tag (h3 for example)

File: styles/App.scss [hunk 2]
```diff
@@ -1,3 +1,30 @@
...
+        img {
+            object-fit: contain;
+            height: 32vh;
+            flex-grow: 4;
+        }
+
+        .title {
+            background-color: #78736F;
+            padding: 10px;
+            // text-overflow: ellipsis;
```
Comment (Veryyapeee) [inline styles/App.scss]: if we don't need piece of code, we should remove it

File: src/components/TileComponents.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { render } from '../shared/dom';
```
Comment (Veryyapeee) [inline src/components/TileComponents.js]: This should be a single, TileComponent
The TileComponents would be an separate component with scroll-like logic

File: styles/App.scss [hunk 4]
```diff
@@ -1,3 +1,30 @@
 body {
     font-family: 'Montserrat', sans-serif;
 }
+
+
+.movie-tiles {
```
Comment (Veryyapeee) [inline styles/App.scss]: We should put styles per component
So it should be tileComponent.scss in folder when TileComponent.js is
Also any color, background-color etc. should be taken from variables

File: src/components/TileComponents.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+export const TileComponent = ({ title, image }) => {
+    return `
+    <div class="movie-tile">
```
Comment (Veryyapeee) [inline src/components/TileComponents.js]: Also I remember that we were talking about this "hover" effect on Tile, when we hover over and we see some details about the certain asset, this should be implemented in scope of this task too

**Review by Veryyapeee — CHANGES_REQUESTED** [review]

**Comment by Veryyapeee** [comment]
Also it's weird that the height of the title "section" doesn't have fixed height, it should be, if text doesn't fit, we should cut it and add "..." at the end

File: src/components/TileComponents.js (hunk 1 above)
Comment (N00BKU) [reply src/components/TileComponents.js]: paragraph tag has been changed for H3 tag, I didn't edit the style for this tag, we can do it later in SCSS file

File: styles/App.scss (hunk 2 above)
Comment (N00BKU) [reply styles/App.scss]: code has been changed

File: src/components/TileComponents.js (hunk 3 above)
Comment (N00BKU) [reply src/components/TileComponents.js]: I edited code, it renders now a single tile, which contains title and image

File: styles/App.scss (hunk 4 above)
Comment (N00BKU) [reply styles/App.scss]: I've made a new file for this in the same folder, now styles are in good place

File: src/components/TileComponents.js (hunk 5 above)
Comment (N00BKU) [reply src/components/TileComponents.js]: hover effect has been added, now after hover on movie image we can see movie details. I need to know whether I should style it now or later?

Comment (N00BKU) [reply src/components/TileComponents.js]: hover effect has been added, now after hover on movie image we can see movie details. I need to know whether I should style it now or later.

File: src/screens/Home.js [hunk 6]
```diff
@@ -1,9 +1,22 @@
...
+        <template id="home-movie-tiles"></template>
+    </div>
 `;
 
-export const HomeScreen = ({ renderOn, options }) => {
+export const HomeScreen = async ({ renderOn, options }) => {
     render({on: renderOn, html: template});
-};
+
+    const movies = await ApiProvider.search('Fast & Furious');
```
Comment (Veryyapeee) [inline src/screens/Home.js:17]: Please add TODO Comment here that this is temporary for now :D

File: src/components/TileComponents.js (hunk 5 above)
Comment (Veryyapeee) [reply src/components/TileComponents.js]: I can't see anything while I'm hovering over :O, and yea, it should be styled in this PR

**Comment by Veryyapeee** [comment]
Generally LGTM, just add these hover styles and I think that you are good to go 8)

**Review by Veryyapeee — APPROVED** [review]

**Review by Mariusz-web — APPROVED** [review]
good

**Comment by Veryyapeee** [comment]
Also we need to add the eventListener for div with onCLick, which will be redirecting to to AssetDetails screen with ID in params

It's waiting for addEventListener utility function, so add TODO comment about it and you can create task for that, for the future :D

File: src/screens/Home.js [hunk 7]
```diff
@@ -1,9 +1,22 @@
...
+
+    const my_movie = movies.results[1];
+    const movie = await ApiProvider.getTitleDetails(my_movie.id);
+    console.log(movie);
```
Comment (Veryyapeee) [inline src/screens/Home.js]: remove console.log

File: src/components/Tiles/MovieTile.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import { render } from '../../shared/dom';
+
+export const MovieTile = ({ renderOn, movie }) => {
```
Comment (Veryyapeee) [inline src/components/Tiles/MovieTile.js]: Generally, the convention should be

For screens we are using the render function, for reusable component, we just need to return the plain html

that way we don't need to pass an ID (which would be painfull), but we would inject it in the html

Ex.

`<div> ${MovieTIle({...someArguments})} </div>` ~ now when MovieTile is returning the html template, it's being equal to just writing this html between these tags, and we have to write it that way, when we would inject like dynamic number of these components

**Review by Veryyapeee — CHANGES_REQUESTED** [review]

**Comment by maciejopalinski** [comment]
@N00BKU hey this is a pretty urgent task because it locks the progress for many other tasks that depend on MovieTile. when will it be done?

**Comment by N00BKU** [comment]
@PoProstuMieciek Hey, i was outside the home at this weekend, very sorry for being late with task. It should be finished by tomorrow.

File: src/screens/Home.js (hunk 7 above)
Comment (N00BKU) [reply src/screens/Home.js]: Done

File: src/components/Tiles/MovieTile.scss [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+        width: 100%;
+        padding: 10px 5px;
+        box-sizing: border-box;
+        font-family: Verdana, Geneva, Tahoma, sans-serif;
+    }
+
+
+}
+
+@media all and (max-width:425px) {
```
Comment (Veryyapeee) [inline src/components/Tiles/MovieTile.scss]: We have media queries defined with mixins in project, use them

File: src/components/Tiles/MovieTile.scss [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+    text-align: center;
+    background-color: $btn-color;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    color: $font-primary-color;
+    width: 100%;
+    padding: auto;
+    box-sizing: border-box;
+    font-family: Verdana, Geneva, Tahoma, sans-serif;
```
Comment (Veryyapeee) [inline src/components/Tiles/MovieTile.scss]: This should be also in variables.scss

File: src/components/Tiles/MovieTile.js (hunk 8 above)
Comment (Veryyapeee) [reply src/components/Tiles/MovieTile.js]: Up, this is not done still

File: src/screens/Home.js (hunk 6 above)
Comment (Veryyapeee) [reply src/screens/Home.js:17]: Up

**Review by Veryyapeee — CHANGES_REQUESTED** [review]

File: src/components/TileComponents.js (hunk 5 above)
Comment (N00BKU) [reply src/components/TileComponents.js]: Hover is working

File: src/screens/Home.js (hunk 6 above)
Comment (N00BKU) [reply src/screens/Home.js:17]: Done

File: src/components/Tiles/MovieTile.js (hunk 8 above)
Comment (N00BKU) [reply src/components/Tiles/MovieTile.js]: I fixed it now

File: src/components/Tiles/MovieTile.scss (hunk 9 above)
Comment (N00BKU) [reply src/components/Tiles/MovieTile.scss]: Now code is using them

File: src/components/Tiles/MovieTile.scss (hunk 10 above)
Comment (N00BKU) [reply src/components/Tiles/MovieTile.scss]: I added title font in variables.scss

File: src/components/Tiles/MovieTile.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+export const MovieTile = ({ movie }) => {
+    return  `
+    <div id="es" class="movie-tile">
```
Comment (Veryyapeee) [inline src/components/Tiles/MovieTile.js]: Remove this ID if not needed

File: src/components/Tiles/MovieTile.scss [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+            text-overflow: ellipsis;
+        }
+
+        .movie-details:hover {
+            opacity: 90%;
+        }
+    }
+
+    .title {
+        position: static;
```
Comment (Veryyapeee) [inline src/components/Tiles/MovieTile.scss]: you don't need this position: static;

File: src/components/Tiles/MovieTile.js (hunk 11 above)
Comment (N00BKU) [reply src/components/Tiles/MovieTile.js]: done

File: src/components/Tiles/MovieTile.scss (hunk 12 above)
Comment (N00BKU) [reply src/components/Tiles/MovieTile.scss]: position has been deleted for now

**Review by Veryyapeee — APPROVED** [review]

**Pull request merged**